providers/agent/mcorpc/client/options.go

```diff
@@ -148,6 +148,21 @@ func (o *RequestOptions) ConfigureMessage(msg *choria.Message) (err error) {
 		return err
 	}
 
+	// calculate a TTL for messages when we have batches and when using cached transports,
+	// we need to avoid 2FA interactions for the full duration of the message:
+	//
+	// (TTL + DiscoveryTimeout + Timeout) * batches
+	//
+	// We have to allow TTL per batch since the last batch will get it much
+	if msg.IsCachedTransport() && o.BatchSize != len(o.Targets) {
+		batches := int(math.Ceil(float64(len(o.Targets)) / float64(o.BatchSize)))
+
+		msg.TTL = batches * (msg.TTL + int(o.DiscoveryTimeout.Seconds()) + int(o.Timeout.Seconds()))
+		if msg.TTL > int((24 * time.Hour).Seconds()) {
+			return fmt.Errorf("cached transport TTL is unreasonably long")
```

Comment:
@snakeb1t I added this check here, the reaon is that the TTL is the window of packet replay - someone capturing your signed packet can replay it within TTL so generally you want it short but you definitely do not want it very long.

Will this bite you do you think?

Reply:
I think it will because I have a large cluster and a use case where I need small batches to run. Can we clamp to 24 hours? Or better yet, make it a config value?

Reply:
Yeah so as is it’s a day. Good yeah? Rather add a config option when we know we need it

Reply:
Yeah, having a max of one day is good enough for me. I don't expect batches to run longer than that